src/components/reusable/datePicker/datepicker.ts

```diff
@@ -79,135 +146,295 @@ export class DatePicker extends FormMixin(LitElement) {
+    const descriptionId = this.nameAttr ?? '';
+    const placeholder = getPlaceholder(this.dateFormat);
+
     return html`
-      <div class="date-picker" ?disabled=${this.disabled}>
-        <label class="label-text" for=${this.name} ?disabled=${this.disabled}>
+      <div class=${classMap(this.getDatepickerClasses())}>
+        <label
+          class="label-text"
+          for=${anchorId}
+          ?disabled=${this.datePickerDisabled}
+        >
           ${this.required
             ? html`<abbr
                 class="required"
-                title=${this._textStrings.requiredText}
-                aria-label=${this._textStrings.requiredText}
+                title=${this._textStrings?.requiredText || 'Required'}
+                role="img"
+                aria-label=${this._textStrings?.requiredText || 'Required'}
                 >*</abbr
               >`
             : null}
           <slot></slot>
         </label>
 
-        <div
-          class="${classMap({
-            'input-wrapper': true,
-          })}"
-        >
+        <div class="input-wrapper">
           <input
-            class="${classMap({
-              'size--sm': this.size === 'sm',
-              'size--lg': this.size === 'lg',
-            })}"
-            datePickerType=${this.datePickerType}
-            type=${this.datePickerType === DATE_PICKER_TYPES.WITHITIME
-              ? 'datetime-local'
-              : 'date'}
-            id=${this.name ? this.name : 'datepicker-1'}
-            name=${this.name}
-            value=${this.value}
+            class="input-custom"
+            type="text"
+            id=${anchorId}
+            name=${this.nameAttr}
+            placeholder=${placeholder}
+            ?disabled=${this.datePickerDisabled}
             ?required=${this.required}
-            ?disabled=${this.disabled}
-            ?invalid=${this._isInvalid}
-            aria-invalid=${this._isInvalid}
-            aria-describedby=${this._isInvalid
-              ? 'error'
-              : this.warnText !== '' && !this._isInvalid
-              ? 'warning'
-              : ''}
-            min=${ifDefined(this.minDate)}
-            max=${ifDefined(this.maxDate)}
-            step=${ifDefined(this.step)}
-            @input=${(e: any) => this.handleInput(e)}
+            aria-invalid=${this._isInvalid ? 'true' : 'false'}
           />
+          <span class="icon">${unsafeSVG(calendarIcon)}</span>
         </div>
-        ${this.caption !== ''
-          ? html` <div class="caption">${this.caption}</div> `
-          : null}
-        ${this._isInvalid
-          ? html`
-              <div id="error" class="error">
-                ${this.invalidText || this._internalValidationMsg}
-              </div>
-            `
-          : null}
-        ${this.warnText !== '' && !this._isInvalid
-          ? html`<div id="warning" class="warn">${this.warnText}</div>`
-          : null}
+
+        ${this.caption
```

Comment:
When I click on the caption or the error message, it also opens the picker.

Reply:
this one is a bit elusive. i don't know if this is a weird side-effect of flatpickr as well. i've tried adjusting the html and pointer-events but not currently working as expected. still working on it

Reply:
✅